File: main/n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal-2.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        acc = []

        def dfs(node):
            for child in node.children:
                dfs(child)
            acc.append(node.val)

        if root is not None:
            dfs(root)

        return acc

File: main/n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal-2.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
    public List<Integer> postorder(final Node root) {
        final List<Integer> acc = new ArrayList<>();
        if (root != null) dfs(acc, root);
        return acc;
    }

    private static void dfs(final List<Integer> acc, final Node node) {
        for (final Node child : node.children) dfs(acc, child);
        acc.add(node.val);
    }
}
